File: javascript/palindromo.js
function normalizar(texto) {
    return texto.toLowerCase().replace(/[^a-z0-9]/g, "");
}

function ehPalindromo(texto) {
    const normalizado = normalizar(texto);
    const invertido = normalizado.split("").reverse().join("");
    return normalizado === invertido;
}

function main() {
    const texto = process.argv[2];

    if (!texto) {
        console.log('Uso: node palindromo.js "texto a verificar"');
        process.exit(1);
    }

    const resultado = ehPalindromo(texto);

    console.log(`Texto original:  ${texto}`);
    console.log(`Texto comparado: ${normalizar(texto)}`);
    console.log(resultado ? "É palíndromo!" : "Não é palíndromo.");

    process.exit(resultado ? 0 : 1);
}

main();

File: java/Palindromo.java
import java.io.PrintStream;
import java.nio.charset.StandardCharsets;

public class Palindromo {

    static String normalizar(String texto) {
        StringBuilder resultado = new StringBuilder();
        for (char c : texto.toLowerCase().toCharArray()) {
            if (Character.isLetterOrDigit(c)) {
                resultado.append(c);
            }
        }
        return resultado.toString();
    }

    static boolean ehPalindromo(String texto) {
        String normalizado = normalizar(texto);
        String invertido = new StringBuilder(normalizado).reverse().toString();
        return normalizado.equals(invertido);
    }

    public static void main(String[] args) {
        System.setOut(new PrintStream(System.out, true, StandardCharsets.UTF_8));

        if (args.length != 1) {
            System.out.println("Uso: java Palindromo \"texto a verificar\"");
            System.exit(1);
        }

        String texto = args[0];
        boolean resultado = ehPalindromo(texto);

        System.out.println("Texto original:  " + texto);
        System.out.println("Texto comparado: " + normalizar(texto));
        System.out.println(resultado ? "É palíndromo!" : "Não é palíndromo.");

        System.exit(resultado ? 0 : 1);
    }
}
